src/main/java/br/com/budgetboys/pf_finance_service/adapters/outbound/service/income/IncomeService.java

```diff
@@ -13,31 +17,39 @@ public class IncomeService {
 
     private final IncomeRepository incomeRepository;
 
+    @Autowired
+    private IncomeMapper incomeMapper;
+
     public IncomeService (IncomeRepository incomeRepository){
         this.incomeRepository = incomeRepository;
     }
     
-    public Income saveIncome(Income income){
+    public IncomeResponseDTO saveIncome(IncomeCreateDTO income){
 
         if(income.getAmount() < 0){
             throw new IllegalArgumentException("The income amount cannot be negative");
         }
-        return incomeRepository.save(income);
+
+        Income incomeEntity = this.incomeMapper.requestToEntity(income);
+
+        Income jpaIncomeEntity = this.incomeRepository.save(incomeEntity);
+
+        return this.incomeMapper.entityToResponse(jpaIncomeEntity);
     }
 
-    public Income findIncomeById(UUID id){
-        Income income = incomeRepository.findById(id);
+    public IncomeResponseDTO findIncomeById(UUID id){
+        Income income = this.incomeRepository.findById(id);
         if(income == null){
             throw new IllegalArgumentException("Income Id: " +id+ "not found");
         }
-        return income;
+        return this.incomeMapper.entityToResponse(income);
     }
 
-    public List<Income> getAllIncomes(){
-        return incomeRepository.findAll();
+    public List<IncomeResponseDTO> findAllIncomes(){
+        return this.incomeRepository.findAll();
     }
```

Comment:
Aqui tu aplicaria desta forma:
```suggestion
public List<IncomeResponseDTO> findAllIncomes() {
    return this.incomeRepository.findAll()
        .stream()
        .map(incomeMapper::entityToResponse)
        .collect(Collectors.toList());
}
```